src/diffusers/utils/import_utils.py

```diff
@@ -369,6 +369,32 @@ def is_torchsde_available():
     return _torchsde_available
 
 
+def is_torch_fp16_available(device):
+    if not is_torch_available():
+        return False
+
+    import torch
+
+    try:
+        x = torch.zeros((2, 2), dtype=torch.float16).to(device)
```

Comment:
a bit worried here that this function is not explicit enough, *e.g.* what if the user has the nvidia-driver incorrectly installed, then `x = torch.zeros((2, 2), dtype=torch.float16).to(device)` will fail but not because fp16 is not available. 

Can we be more specific here when catching the error? Maybe just an exception related to "Format is not supported"?